File: Benchmarking/src/gen175.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  double f12(const double &v0) {
    double v2 = v0;
    double v1 = v0;
    double v23 = v2;
    double v15 = v1;
    double v17 = v15;
    double v16 = v0;
    double v22 = v17;
    double v19 = v16;
    v19 = v22;
    double v24 = v23;
    v15 = v24;
    double v51 = v19;
    double v40 = v51;
    return v40;
  }
  __attribute__((noinline))
  double f0(const std::vector<std::vector<s0>> &v0, const double &v1) {
    const double v4 = v1 + v1;
    const double v3 = f12(v1);
    const double v2 = f12(v4);
    const double v12 = v2 / v3;
    double v14 = v3;
    double v16 = v4;
    double v23 = v14;
    const double v29 = v2 - v16;
    v14 = v12;
    double v261 = v29;
    const double v253 = v261 * v23;
    return v253;
  }
  int main() {
    std::vector<std::vector<s0>> v0({ { { { { 0.0 } }, { { 1.0 }, { 2.0 } } } }, { { { { 3.0 } }, { { 4.0 }, { 5.0 } } } }, { { { { 6.0 } }, { { 7.0 }, { 8.0 } } } } });
    double v1(9.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen175.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  func f12(_ v0: Double) -> Double {
    var v2: Double = v0
    var v1: Double = v0
    var v23: Double = v2
    var v15: Double = v1
    var v17: Double = v15
    var v16: Double = v0
    var v22: Double = v17
    var v19: Double = v16
    v19 = v22
    var v24: Double = v23
    v15 = v24
    var v51: Double = v19
    var v40: Double = v51
    return v40
  }
  @inline(never)
  func f0(_ v0: [[s0]], _ v1: Double) -> Double {
    let v4: Double = v1 + v1
    let v3: Double = f12(v1)
    let v2: Double = f12(v4)
    let v12: Double = v2 / v3
    var v14: Double = v3
    var v16: Double = v4
    var v23: Double = v14
    let v29: Double = v2 - v16
    v14 = v12
    var v261: Double = v29
    let v253: Double = v261 * v23
    return v253
  }
  func benchmark() {
    let v0: [[s0]] = [[s0(p0: [[0.0]], p1: [[1.0], [2.0]])], [s0(p0: [[3.0]], p1: [[4.0], [5.0]])], [s0(p0: [[6.0]], p1: [[7.0], [8.0]])]]
    let v1: Double = 9.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 9.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
